control-plane/agents/src/bin/ha/cluster/switchover.rs

```diff
@@ -406,50 +408,57 @@ impl SwitchOverRequest {
-                    // otherwise this means we were able to reconnect even before we got called!
-                    if !self.reuse_existing {
-                        info!(volume.uuid=%self.volume_id, "Moving Switchover request to Errored state");
-                        self.set_stage(Stage::Errored);
-                        Err(anyhow!("HA Node agent could not lookup old Nvme path"))
-                    } else {
-                        Ok(())
-                    }
+
+        let Some(new_path) = self.new_path.clone() else {
+            return Err(anyhow!("Could not to get new nexus target for the volume"));
+        };
+
+        let replace_request = ReplacePath::new(
+            self.existing_nqn.clone(),
+            new_path,
+            self.publish_context.clone(),
+        );
+        let client = NodeAgentClient::new(uri, None).await;
+        match client.replace_path(&replace_request, None).await {
+            Ok(_) => Ok(()),
+            Err(error) if error.kind == ReplyErrorKind::FailedPrecondition => {
+                warn!(path=%self.existing_nqn, "HA Node agent could not find failed Nvme path");
+                // otherwise this means we were able to reconnect even before we got called!
+                if !self.reuse_existing {
+                    info!(volume.uuid=%self.volume_id, "Moving Switchover request to Errored state");
+                    self.set_stage(Stage::Errored);
+                    Err(anyhow!("HA Node agent could not lookup old Nvme path"))
+                } else {
+                    Ok(())
                 }
-                Err(error)
-                    if matches!(
-                        error.kind,
-                        ReplyErrorKind::Aborted
-                            | ReplyErrorKind::Timeout
-                            | ReplyErrorKind::NotFound
-                    ) =>
-                {
-                    info!(volume.uuid=%self.volume_id, "Retrying Republish without older target reuse");
-                    self.set_reuse_existing(false);
-                    self.set_stage(Stage::RepublishVolume);
-                    return Err(anyhow!("Nvme path replacement failed with older target"));
+            }
+            Err(error)
+                if matches!(
+                    error.kind,
+                    ReplyErrorKind::Aborted
+                        | ReplyErrorKind::Timeout
+                        | ReplyErrorKind::NotFound
+                        | ReplyErrorKind::DeadlineExceeded
+                ) =>
+            {
+                if !matches!(error.kind, ReplyErrorKind::NotFound) {
+                    warn!(path=%self.existing_nqn, "HA Node agent failed to replace path within timeout");
+                    if self.retry_count < REPLACE_PATH_DEADLINE_RETRIES {
+                        return Err(anyhow!("Nvme path deadline exceeded"));
+                    }
```

Comment:
If I read this right, It seems we'd end up retrying when we get NotFound, is that intended?

Reply:
NotFound is hit when the new path is not present for some odd reason (maybe crash after previous republish?).
In this case there's no point retrying to connect, better to go back to republish?

Reply:
ok, so this logic is indeed doing republish for this(NotFound) case. Sounds alright.